### PR TITLE
(PDK-1007) implement enough to support purge=>true 

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

**Comment by DavidS** [comment]
The code coverage failure seems to be only empty lines.

**Comment by DavidS** [comment]
rebased to master

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
+This is an implementation of the [Resource API](https://github.com/puppetlabs/puppet-specifications/blob/master/language/resource-api/README.md) proposal. Find a working example of a new-style provider in the [experimental puppetlabs-apt branch](https://github.com/DavidS/puppetlabs-apt/blob/resource-api-experiments/lib/puppet/provider/apt_key2/apt_key2.rb). There is also the corresponding [type](https://github.com/DavidS/puppetlabs-apt/blob/resource-api-experiments/lib/puppet/type/apt_key2.rb), [provider](https://github.com/DavidS/puppetlabs-apt/blob/resource-api-experiments/lib/puppet/provider/apt_key2/apt_key2.rb), and [new unit tests](https://github.com/DavidS/puppetlabs-apt/blob/resource-api-experiments/spec/unit/puppet/provider/apt_key2/apt_key2_spec.rb) for 100% coverage.
```
Comment (da-ar) [inline README.md]: Looks like another reference on [line 179](https://github.com/puppetlabs/puppet-resource_api/pull/95/commits/f0605caa4794941bcaf4724473121dc4948db15f#diff-04c6e90faac2675aa89e2176d2eec7d8L179) that requires updating as well

Comment (DavidS) [reply README.md]: indeed!

File: spec/acceptance/composite_namevar_spec.rb [hunk 2]
```diff
@@ -32,12 +32,12 @@
       expect(stdout_str.strip).to match %r{ensure\s*=> \'present\'}
       expect(status).to eq 0
     end
-    it 'returns the match if title matches a namevar value' do
-      stdout_str, status = Open3.capture2e("puppet resource #{common_args} composite_namevar php")
-      expect(stdout_str.strip).to match %r{^composite_namevar \{ \'php\'}
-      expect(stdout_str.strip).to match %r{ensure\s*=> \'present\'}
-      expect(status).to eq 0
-    end
+    # it 'returns the match if title matches a namevar value' do
```
[the remark this answers is not in the document]
Comment (DavidS) [reply spec/acceptance/composite_namevar_spec.rb]: re-activated them

**Comment by DavidS** [comment]
Addressed comments and rebased

**Pull request merged**